extmod/modnetwork.c

```diff
@@ -89,9 +89,52 @@ STATIC mp_obj_t network_route(void) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_0(network_route_obj, network_route);
 
+
+char mod_network_country_code[2] = "XX";
+
+STATIC mp_obj_t network_country(size_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return mp_obj_new_str(mod_network_country_code, 2);
+    } else {
+        size_t len;
+        const char *str = mp_obj_str_get_data(args[0], &len);
+        if (len != 2) {
+            mp_raise_ValueError(NULL);
+        }
+        mod_network_country_code[0] = str[0];
+        mod_network_country_code[1] = str[1];
+        return mp_const_none;
+    }
+}
+// TODO: Non-static to allow backwards-compatible pyb.country.
+MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mod_network_country_obj, 0, 1, network_country);
+
+#ifndef MICROPY_PY_NETWORK_DEFAULT_HOSTNAME
+#error "MICROPY_PY_NETWORK_DEFAULT_HOSTNAME must be set in mpconfigport.h or mpconfigboard.h"
```

Comment:
how about naming this `MICROPY_PY_NETWORK_HOSTNAME_DEFAULT`?  the point being that it's a configuration of `network.hostname`